### PR TITLE
Add transaction manager to passwordlessLogin and login

### PR DESCRIPTION
**Pull request by luisrudge**
This fixes an issue that state/nonce were not being auto generated in popup mode.
fix https://github.com/auth0/auth0.js/issues/730

### CODE REVIEW

**Comment by Neaox** [comment]
Any updates as to when this might be released?

**Comment by luisrudge** [comment]
We still have a few outstanding issues we want in the next release. It's either tomorrow or next week.

File: src/web-auth/index.js [hunk 1]
```diff
@@ -651,18 +666,33 @@ WebAuth.prototype.login = function(options, cb) {
  * @param {crossOriginLoginCallback} cb Callback function called only when an authentication error, like invalid username or password, occurs. For other types of errors, there will be a redirect to the `redirectUri`.
  */
 WebAuth.prototype.passwordlessLogin = function(options, cb) {
+  var params = objectHelper
```
[the remark this answers is not in the document]
Comment (luisrudge) [reply src/web-auth/index.js]: We use that a bunch of different places, so I feel this should be tackled in a bigger refactor on how we handle transactions. Tracking: https://github.com/auth0/auth0.js/issues/740

File: test/web-auth/web-auth.test.js [hunk 2]
```diff
@@ -1806,7 +1810,11 @@ describe('auth0.WebAuth', function() {
           credentialType: 'http://auth0.com/oauth/grant-type/passwordless/otp',
           realm: 'email',
           username: 'the@email.com',
-          otp: '123456'
+          otp: '123456',
+          clientID: '...',
```
[the remark this answers is not in the document]
Comment (luisrudge) [reply test/web-auth/web-auth.test.js]: We allow people to override most of the options in all the methods, that's why I added all the options that you can pass to `authorize`, except responseMode.

[the remark this answers is not in the document]
Comment (lbalmaceda) [reply test/web-auth/web-auth.test.js]: Alright with (1). Solve (2) and I'll approve.

[the remark this answers is not in the document]
Comment (luisrudge) [reply test/web-auth/web-auth.test.js]: done

Comment (luisrudge) [inline test/web-auth/web-auth.test.js]: done

File: test/web-auth/web-auth.test.js [hunk 3]
```diff
@@ -1754,7 +1754,8 @@ describe('auth0.WebAuth', function() {
         clientID: '...',
         redirectUri: 'http://page.com/callback',
         responseType: 'code',
-        _sendTelemetry: false
+        _sendTelemetry: false,
+        nonce: 'the-nonce'
```
Comment (lbalmaceda) [inline test/web-auth/web-auth.test.js]: You shouldn't be setting the nonce value yourself. The PR description clearly states that what was broken is "state/nonce were not being auto generated". Please DO check that the nonce is present but don't manually set it.

Comment (lbalmaceda) [reply test/web-auth/web-auth.test.js]: I think this line is already doing some of that https://github.com/auth0/auth0.js/blob/f6e2227bfe921e30fae73a31b033d12255441d08/test/web-auth/web-auth.test.js#L27

Comment (luisrudge) [reply test/web-auth/web-auth.test.js]: you're right. I switched to id_token since it's the only case where it generates the nonce.

**Pull request merged**